### PR TITLE
feat: Add Minecraft name access HypixelGuild mode

### PR DESCRIPTION
**Pull request by lycorisor**

### CODE REVIEW

File: config/manager.go [hunk 1]
```diff
@@ -42,6 +45,7 @@ func LoadConfig() {
 	}
 
 success:
+	log.Println(color.HiYellowString("Start loading file."))
```
Comment (layou233) [inline config/manager.go:48]: This is a little annoying.

File: main.go [hunk 2]
```diff
@@ -52,17 +52,24 @@ func main() {
...
-		signal.Notify(osSignals, os.Interrupt, syscall.SIGTERM)
-		<-osSignals
-		// stop the program
-		// sometimes after the program exits on Windows, the ports are still occupied and "listening".
-		// so manually closes these listeners when the program exits.
-		for _, listener := range service.ListenerArray {
-			if listener != nil { // avoid null pointers
-				listener.Close()
+	osSignals := make(chan os.Signal, 1)
+	signal.Notify(osSignals, os.Interrupt, os.Kill, syscall.SIGTERM, syscall.SIGHUP)
```
Comment (layou233) [inline main.go:56]: os.Kill seems like a uncatchable signal and I have removed it in previous commit. Please check it again.

File: config/manager.go [hunk 3]
```diff
@@ -104,14 +108,27 @@ func LoadLists(isReload bool) bool {
...
 	} else {
 		Lists = make(map[string]*set.StringSet, l) // map size init
 		for k, v := range Config.Lists {
 			// log.Println("List: Loading", k, "value:", v)
-			list := set.NewStringSetFromSlice(v)
-			Lists[k] = &list
+			switch ListsModeTable[k] {
+			case "HypixelGuild":
+				// v[0] A Player Name in Guild
+				// v[1] Hypixel ApiKey
```
Comment (layou233) [inline config/manager.go:121]: This design is..... not graceful enough.

My opinion is add some new options to NameAccess object.

Btw, the new HypixelGuild mode is LGTM.

File: config/manager.go [hunk 4]
```diff
@@ -158,3 +175,77 @@ func MonitorConfig(watcher *fsnotify.Watcher) error {
...
+	cli := http.DefaultClient
+	req, err := http.NewRequest("POST", "https://api.mojang.com/profiles/minecraft", strings.NewReader("[\""+info[0]+"\"]"))
+	if err != nil {
+		return err, nil
+	}
+	resp, err := cli.Do(req)
+	if err != nil {
+		return err, nil
+	}
+	body, err := io.ReadAll(resp.Body)
```
Comment (layou233) [inline config/manager.go:202]: Do you need to close the request body here?

**Review by layou233 — CHANGES_REQUESTED** [review]

File: config/manager.go (hunk 1 above)
Comment (lycorisor) [reply config/manager.go:48]: Without prompting this, the player has no way of knowing the status of the reload

**Pull request closed without merging**